### ISSUE TITLE
Appbase - Missing Active Authority

### ISSUE THREAD

**Issue by mvandeberg** [steemit/steem#2102]
I have not yet tested with other transfer/active auth operations. I am in process of investigating the issue but am wanting to document what I have found thus far.

Attempting to transfer SBD to savings in dev results in the error: "Missing Active Authority".

The exact API call was: 
`{"id":"673817549053440","jsonrpc":"2.0","method":"call","params":["condenser_api","broadcast_transaction_synchronous",[{"ref_block_num":63503,"ref_block_prefix":3678793423,"expiration":"2018-02-09T17:33:21","operations":[["transfer_to_savings",{"from":"vandeberg","to":"vandeberg","amount":"0.001 SBD","memo":""}]],"extensions":[],"signatures":["202007d6d8328bb9569272ffde0b375b3fc48beac64dea937974c57383cf462987183bf2cda74e01e32c07e5fabc93d77729df153242ff27138cafd9cc3de64d89"]}]]}`

I converted to the equivalent op using `network_broadcast_api` and it failed. That API call was:

`{"id":"673817549053440","jsonrpc":"2.0","method":"network_broadcast_api.broadcast_transaction_synchronous", "params":{"trx":{"ref_block_num":63503,"ref_block_prefix":61135,"expiration":"2018-02-09T17:33:21","operations":[["transfer_to_savings",{"from":"vandeberg","to":"vandeberg","amount":["1",3,"@@000000013"],"memo":""}]],"extensions":[],"signatures":["202007d6d8328bb9569272ffde0b375b3fc48beac64dea937974c57383cf462987183bf2cda74e01e32c07e5fabc93d77729df153242ff27138cafd9cc3de64d89"]},"max_block_age":-1}}`

I also sent broadcast the identical transaction in prod with success. That API call was:

`{"id":"673817549053440","jsonrpc":"2.0","method":"call","params":["network_broadcast_api","broadcast_transaction_synchronous",[{"ref_block_num":63503,"ref_block_prefix":3678793423,"expiration":"2018-02-09T17:33:21","operations":[["transfer_to_savings",{"from":"vandeberg","to":"vandeberg","amount":"0.001 SBD","memo":""}]],"extensions":[],"signatures":["202007d6d8328bb9569272ffde0b375b3fc48beac64dea937974c57383cf462987183bf2cda74e01e32c07e5fabc93d77729df153242ff27138cafd9cc3de64d89"]}]]}`

The included transaction is [here](https://steemd.com/tx/06e6328fe6cb2a5707ddadca53483798716c0215).

At first I was quite worried. A transaction is being rejected by `develop` but is included in `stable`? It should fork the network. But it doesn't. The savings balance is identical on both dev and prod long after the block was made irreversible. It would seem that when the transaction is applied from the `network_broadcast_api` it fails, but it was applied when the block was pushed and the node did not fork.

**Comment by mvandeberg** [comment]
The same thing occurs with a normal transfer.

**Comment by mvandeberg** [comment]
After much digging and a little bit of oversight I identified the bug as incorrectly defining `ref_block_prefix` as 16 bit in `legacy_signed_transaction`.

This can be seen in my transactions above. To get the correct asset conversion above, I dumped the transaction in `network_broadcast_api`. However, at that point the reference block prefix had already been truncated and resulted in the same wrong behavior.
